tracpro/polls/models.py

```diff
@@ -472,7 +472,21 @@ def get_responses(self, region=None, include_subregions=True,
                 responses = responses.filter(contact__region=region)
         if not include_empty:
             responses = responses.exclude(status=Response.STATUS_EMPTY)
-        return responses.select_related('contact')
+
+        responses = responses.select_related('contact')
+
+        # Remove duplicate resopnse from the same user on the same day
+        response_pk_list, response_tuple_set = [], set()
```

Comment:
My tendency would be to use variable names like "response_pks" and "response_tuples" rather than building the Python type into the variable name. Again, for readability.

This one is fuzzier, though. There are definitely times when I'd want to be explicit in the variable names about what types things are, to avoid confusion or even possible errors when using them. And this case, where we have two collections being used in the same code that are of different types, might be such a time.  Though maybe we could have both readability and less chance of forgetting what type things are if we just made both collections sets here... the collection of PKs doesn't need to be a set, but it wouldn't really hurt.  Sometimes it's worth sacrificing a micro-optimization for readability and reducing the chance of someone accidentally breaking the code when making later changes.

To summarize ... I'd like something done here so we end up with better variable names, but there are multiple options I'd be okay with to address it.